runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -307,6 +318,97 @@ int32_t TR_EscapeAnalysis::perform()
 #endif
 
    cost = performAnalysisOnce();
+
+   if (!_callsToProtect->empty() && manager()->numPassesCompleted() < _maxPassNumber)
+      {
+      TR::CFG *cfg = comp()->getFlowGraph();
+      TR::Block *block = NULL;
+      TR::TreeTop *lastTree = comp()->findLastTree();
+      TR_EscapeAnalysisTools tools(comp());
+      RemainingUseCountMap *remainingUseCount = new (comp()->trStackMemory()) RemainingUseCountMap(RemainingUseCountMapComparator(), comp()->trMemory()->currentStackRegion());
+      for (TR::TreeTop *tt = comp()->findLastTree(); tt != NULL; tt = tt->getPrevTreeTop())
+         {
+         TR::Node *node = tt->getNode();
+         if (node->getOpCodeValue() == TR::BBEnd)
+            block = node->getBlock();
+         else if (node->getOpCode().isCheck() || node->getOpCode().isAnchor() || node->getOpCodeValue() == TR::treetop || node->getOpCode().isStoreDirect())
+            node = node->getFirstChild();
```

Comment:
Do you also need to account for the fact that a store node may be under a treetop node that in turn may have a first child in this condition ?